### PR TITLE
vim-patch:7.4.2230

### LINKED ISSUE #712: Use 'smartcase' with :tselect/:tj etc
I think a common setting is `:set ignorecase smartcase`, which ignores case for lowercase patterns, but not for mixed case.

While this works for searching in general, it's not honored with `:tj`: `:tj Foo` will also present a tag `foo`, even though `Foo` exists.
Even if `Foo` would not exist, `:tj Foo` should not go to `foo`.

### PR DESCRIPTION
**Pull request by lonerover**
Problem:    There is no equivalent of 'smartcase' for a tag search.
Solution:   Add value "followscs" and "smart" to 'tagcase'. (Christian
            Brabandt, closes vim/vim#712) Turn tagcase test into new style.

https://github.com/vim/vim/commit/66e29d7112e437b2b50efe1f82c7e892736d23e4

### CODE REVIEW

**Comment by lonerover** [comment]
As the tagcase test has been translated into lua, do we still need the new style test?

**Comment by justinmk** [comment]
Sure, when vim gets new-style tests it doesn't hurt to add them. Even if they're mostly redundant with the lua tests.

**Comment by lonerover** [comment]
Thank you. I'll add the new style test later.

**Comment by lonerover** [comment]
The new style test need lambda support. Waiting for the merge of https://github.com/neovim/neovim/pull/5771 now.

**Comment by justinmk** [comment]
#5771 was merged

**Comment by lonerover** [comment]
ready now

File: src/nvim/testdir/test_tagcase.vim [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+func Test_tagcase()
+  call writefile(["Bar\tXtext\t3", "Foo\tXtext\t2", "foo\tXtext\t4"], 'Xtags')
+  set tags=Xtags
+  e Xtext
+
+  for &ic in [0, 1]
+    for &scs in [0, 1]
+      for &g:tc in ["followic", "ignore", "match", "followscs", "smart"]
+	for &l:tc in ["", "followic", "ignore", "match", "followscs", "smart"]
```
Comment (justinmk) [inline src/nvim/testdir/test_tagcase.vim]: the indentation looks wrong

**Comment by justinmk** [comment]
missing change to version.c

**Comment by lonerover** [comment]
Sorry, version.c has been updated

**Pull request merged**